wallet/src/de/schildbach/wallet/service/WalletTransactionMetadataProvider.kt

```diff
@@ -16,30 +16,43 @@
 
 package de.schildbach.wallet.service
 
+import de.schildbach.wallet.data.AddressMetadataDao
 import de.schildbach.wallet.data.TransactionMetadataDao
+import kotlinx.coroutines.*
 import kotlinx.coroutines.flow.*
+import org.bitcoinj.core.Address
 import org.bitcoinj.core.Coin
 import org.bitcoinj.core.Context
 import org.bitcoinj.core.Sha256Hash
+import org.bitcoinj.core.Transaction
+import org.bitcoinj.script.ScriptPattern
+import org.bitcoinj.utils.Fiat
 import org.bitcoinj.utils.MonetaryFormat
 import org.dash.wallet.common.WalletDataProvider
 import org.dash.wallet.common.data.*
 import org.dash.wallet.common.services.TransactionMetadataProvider
 import org.dash.wallet.common.transactions.TaxCategory
 import org.dash.wallet.common.transactions.TransactionCategory
 import org.dash.wallet.common.transactions.TransactionMetadata
+import org.dash.wallet.common.transactions.TransactionUtils.isEntirelySelf
 import org.slf4j.LoggerFactory
+import java.util.concurrent.Executors
 import javax.inject.Inject
 
 class WalletTransactionMetadataProvider @Inject constructor(
     private val transactionMetadataDao: TransactionMetadataDao,
+    private val addressMetadataDao: AddressMetadataDao,
     private val walletData: WalletDataProvider
 ) : TransactionMetadataProvider {
 
     companion object {
         val log = LoggerFactory.getLogger(WalletTransactionMetadataProvider::class.java)
     }
 
+    private val syncScope = CoroutineScope(
+        Executors.newCachedThreadPool().asCoroutineDispatcher()
+    )
```

Comment:
During the blockchain sync, if there are many transactions this scope, could this result in creating countless threads?  Would it be better to use a fixed pool size or some other scope?

The idea was taken from the CrowdNode module.